### PR TITLE
Sleep only for a specific event

### PR DESCRIPTION
**Pull request by AntonZelenin**

### CODE REVIEW

File: agent/src/agent/api/routes/monitoring.py [hunk 1]
```diff
@@ -76,3 +76,21 @@ def scheduled_script_error(script_name):
...
+@monitoring_bp.route('/monitoring/directory_file_processed/<pipeline_id>', methods=['POST'])
+def directory_file_processed(pipeline_id):
+    pipeline_ = pipeline.repository.get_by_id(pipeline_id)
+    monitoring_.metrics.DIRECTORY_FILE_PROCESSED\
+        .labels(pipeline_.streamsets.url, pipeline_.name)\
+        .inc(1)
+    return jsonify('')
+
+
+@monitoring_bp.route('/monitoring/watermark_delta/<pipeline_id>/<delta>', methods=['POST'])
```
Comment (dariakharlan) [inline agent/src/agent/api/routes/monitoring.py:90]: no need for delta in url

File: agent/src/agent/api/routes/pipeline.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def reset(pipeline_id: str):
     return jsonify('')
 
 
-@pipelines.route('/pipeline-status-change/<pipeline_id>', methods=['POST'])
+@pipelines.route('/pipelines/<pipeline_id>/status', methods=['POST'])
```
Comment (dariakharlan) [inline agent/src/agent/api/routes/pipeline.py:146]: revert

File: agent/src/agent/api/routes/pipeline.py [hunk 3]
```diff
@@ -159,11 +159,17 @@ def pipeline_status_change(pipeline_id: str):
     return jsonify('')
 
 
-@pipelines.route('/pipeline-offset/<pipeline_id>', methods=['POST'])
+@pipelines.route('/pipelines/<pipeline_id>/offset', methods=['POST'])
```
Comment (dariakharlan) [inline agent/src/agent/api/routes/pipeline.py:162]: revert

File: agent/src/agent/pipeline/config/jython_scripts/create_watermark.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        if not slept:
+            # sleep a little bit to let the pipeline finish processing and sending file data
+            time.sleep(int(sdc.userParams['SLEEP_TIME']))
+            slept = True
+
+        res = requests.get(sdc.userParams['PIPELINE_OFFSET_URL'])
+        res.raise_for_status()
+        if not res.json():
+            continue
+        record.value['watermark'] = float(res.json()) + float(sdc.userParams['BUCKET_SIZE_IN_SECONDS'])
```
Comment (dariakharlan) [inline agent/src/agent/pipeline/config/jython_scripts/create_watermark.py:36]: calcualte on the agent side and use logic from kafka

**Review by dariakharlan — CHANGES_REQUESTED** [review]

File: agent/src/agent/api/routes/pipeline.py [hunk 5]
```diff
@@ -167,3 +167,17 @@ def pipeline_offset_changed(pipeline_id: str):
...
     )
     return jsonify('')
+
+
+@pipelines.route('/pipelines/<pipeline_id>/watermark', methods=['GET'])
+@needs_pipeline
+def calculate_watermark(pipeline_id: str):
+    pipeline_ = pipeline.repository.get_by_id(pipeline_id)
+    if not pipeline_.offset:
+        return jsonify('')
```
Comment (dariakharlan) [inline agent/src/agent/api/routes/pipeline.py:177]: maybe we should return something more meaningful so we could at least write it to logs that there is no offset and no watermark/.

**Review by dariakharlan — CHANGES_REQUESTED** [review]

File: agent/src/agent/api/routes/pipeline.py (hunk 5 above)
Comment (AntonZelenin) [reply agent/src/agent/api/routes/pipeline.py:177]: added logs in streamsets

**Review by dariakharlan — APPROVED** [review]

**Pull request merged**